src/equations/serre_green_naghdi_1d.jl

```diff
@@ -846,23 +821,18 @@ For a [`bathymetry_variable`](@ref) the total modified energy has the additional
 `q_global` is a vector of the primitive variables at ALL nodes.
 `cache` needs to hold the SBP operators used by the `solver`.
 """
-function energy_total_modified(q_global,
-                               equations::SerreGreenNaghdiEquations1D,
-                               cache)
+function energy_total_modified!(e, q_global,
```

Comment:
I would like to only need to define the `energy_total_modified!` version for the different equations, but since this is an implementation detail I would not like to present this to the user. Thus, I would like to only document `energy_total_modified`. Now we have a docstring `energy_total_modified` for the function `energy_total_modified!`, which is even weirder. Do you have a better suggestion to achieve this?

Reply:
Ah, I see. I don't like this since Documenter.jl will still show that it's associated with `energy_total_modified!`, isn't it? I don't have a better solution than just replicating the general `energy_total_modified` specialized to each equation and attach the docstring to it.

Reply:
So we see two different options here:

1. Document `energy_total_modified!' properly and mention it in `energy_total_modified` (and maybe export it) (this is basically your suggestion from above)
2. Replicate the general `energy_total_modified` for each equation and attach the docstring to it.

Although I consider this as implementation detail, which I would like to hide from the user I tend to 1. since I really don't like replicating code unnecessarily. Are you fine with that? If so, I think I would not export `energy_total_modified!` (and thus prefix it with DispersiveShallowWater.jl in the docstring).

Reply:
Both options are okay - please choose which one you prefer. The code duplication would only include a small amount of code (allocating the output), so it's not too bad, I think.

Reply:
I opted for 1. in [686ae52](https://github.com/JoshuaLampert/DispersiveShallowWater.jl/pull/146/commits/686ae527fe7e5f4ce253668a7c94e47ab1f99eb7).